src/gui/torrentcreatordlg.h

```diff
@@ -28,52 +28,74 @@
  * Contact : chris@qbittorrent.org
  */
 
-#ifndef CREATE_TORRENT_IMP_H
-#define CREATE_TORRENT_IMP_H
+#ifndef TORRENTCREATORDLG_H
+#define TORRENTCREATORDLG_H
 
-#include "ui_torrentcreatordlg.h"
+#include <QDialog>
+
+namespace Ui
+{
+    class TorrentCreatorDlg;
+}
 
 namespace BitTorrent
 {
     class TorrentCreatorThread;
 }
 
-class TorrentCreatorDlg : public QDialog, private Ui::createTorrentDialog
+class TorrentCreatorDlg: public QDialog
 {
     Q_OBJECT
 
 public:
-    TorrentCreatorDlg(QWidget *parent = 0);
+    TorrentCreatorDlg(QWidget *parent = 0, const QString &defaultPath = QString());
     ~TorrentCreatorDlg();
-    int getPieceSize() const;
 
 public slots:
     void updateProgressBar(int progress);
-    void on_cancelButton_clicked();
-
-protected slots:
-    void on_createButton_clicked();
-    void on_addFile_button_clicked();
-    void on_addFolder_button_clicked();
-    void handleCreationFailure(QString msg);
-    void handleCreationSuccess(QString path, QString branch_path);
-    void setInteractionEnabled(bool enabled);
-    void showProgressBar(bool show);
-    void on_checkAutoPieceSize_clicked(bool checked);
-    void updateOptimalPieceSize();
-    void saveTrackerList();
-    void loadTrackerList();
 
-protected:
-    void closeEvent(QCloseEvent *event);
+private slots:
+    void onCreateButtonClicked();
+    void onAddFileButtonClicked();
+    void onAddFolderButtonClicked();
+    void handleCreationFailure(const QString &msg);
+    void handleCreationSuccess(const QString &path, const QString &branch_path);
 
 private:
+    void dropEvent(QDropEvent *event) override;
+    void dragEnterEvent(QDragEnterEvent *event) override;
+
     void saveSettings();
     void loadSettings();
+    int getPieceSize() const;
+    void showProgressBar(bool show);
+    void setInteractionEnabled(bool enabled);
 
-private:
+    // settings storage
```

Comment:
In this case, all these functions below are overkill. These one-line methods are only used inside the class and only once.